app/models/katello/concerns/redhat_extensions.rb

```diff
@@ -44,16 +44,63 @@ def construct_name(family)
       end
 
       def medium_uri_with_content_uri(host, url = nil)
-        if host.try(:content_source) && (repo_details = kickstart_repo(host))
-          URI.parse(repo_details[:path])
+        if host.try(:content_source) && (repos = distribution_repositories(host)).present?
+          if repos.size == 1
+            # There is exactly 1 ks repo distribution matching the os, arch, release
+            # cv, env for this host. Just get the media path from available repo.
+            URI.parse(repos.first.full_path(host.content_source))
```

Comment:
Why don't we also need to validate here that the distribution repository found matches the medium URI attached to the host? Seems like before we resting on an assumption that the UI or API would handle this properly. As I try to understand this better, I lean towards having some function that captures the "validation" logic so that both locations can make use of it.